### PR TITLE
Add stop in add_mbktomatrixVisitor

### PR DESCRIPTION
**Pull request by olivier-goury**
Adds a stop when the visitor encounters a mapping

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: SofaKernel/framework/sofa/simulation/MechanicalMatrixVisitor.h [hunk 1]
```diff
@@ -213,6 +213,11 @@ class SOFA_SIMULATION_CORE_API MechanicalAddMBK_ToMatrixVisitor : public Mechani
 
         return RESULT_CONTINUE;
     }
+
+    virtual bool stopAtMechanicalMapping(simulation::Node* /*node*/, core::BaseMapping* map)
+    {
```
Comment (EulalieCoevoet) [inline SofaKernel/framework/sofa/simulation/MechanicalMatrixVisitor.h]: Hi, 
Just a tiny comment, you should use "SOFA_UNUSED(node);"

**Comment by guparan** [comment]
Hi @olivier-goury, is this a fix? Could you add a bit of context please? Thanks :-)

File: SofaKernel/framework/sofa/simulation/MechanicalMatrixVisitor.h [hunk 2]
```diff
@@ -213,6 +213,11 @@ class SOFA_SIMULATION_CORE_API MechanicalAddMBK_ToMatrixVisitor : public Mechani
...
+        return !map->f_mapMatrices.getValue();
```
Comment (guparan) [inline SofaKernel/framework/sofa/simulation/MechanicalMatrixVisitor.h]: Should be replaced by `return !map->areMatricesMapped();` ?

**Comment by guparan** [comment]
ping @olivier-goury :-)

**Comment by olivier-goury** [comment]
Hello @guparan, sorry for the delay! 
Yes that is a fix: this mechanical visitor is building the matrices M, B and K of the mechanical system.  If there is a mapping with the value mapMatrices set to 0, then this means the mapping does not wish to map those matrices. This implies the visitor should stop when it encounters such a mapping.

**Comment by olivier-goury** [comment]
Without adding this stop, the default behaviour (define in the parent ) checks if forces are mapped, which is irrelevant for that visitor.

**Comment by guparan** [comment]
[ci-build] [with-scene-tests]

**Pull request merged**